FWCore/Utilities/interface/OffsetToBase.h

```diff
@@ -0,0 +1,55 @@
+#ifndef FWCore_Utilities_OffsetToBase_h
+#define FWCore_Utilities_OffsetToBase_h
+#include <typeinfo>
+
+/*
+ * For any class used in Views, RefToBase, or Ptr,
+ * class template OffsetToBase must be specialized for any class
+ * with multiple inheritance (i.e. with non-zero offset to
+ * at least one base class).
+ * A specialization would look something like this
+ * (in YourClass.h"
+ *
+ * #include "FWCore/Utilities/interface/OffsetToBase.h 
+ * namespace edm {
+ *   template<>
+ *   class OffsetToBase<YourClass> {
+ *     public OffsetToBase() {}
+ *     size_t offsetToBase(std::type_info const& baseTypeInfo) const {
+ *       YourClass object;
```

Comment:
Calling a constructor each time this happens isn't a great idea. How about cheating instead:
``` int const dummy = 0; //Just need an adress of something on the stack
    YourClass const* object = reinterpret_cast<YourClass const*>(&dummy);
```